src/metaxy/metadata_store/postgres.py

```diff
@@ -0,0 +1,972 @@
+                f"PostgresMetadataStore supports only MD5 and SHA256 hash algorithms. "
+                f"Requested: {self.hash_algorithm}"
+            )
+
+    def _get_default_hash_algorithm(self) -> HashAlgorithm:
+        """Get default hash algorithm for PostgreSQL stores.
+
+        Uses MD5 as it's built-in and requires no extensions.
+        For SHA256 support, explicitly set hash_algorithm=HashAlgorithm.SHA256
+        and ensure pgcrypto extension is enabled.
+        For production you must self-install the necessary posgres extensions.
+        """
+        return HashAlgorithm.MD5
+
+    def _create_hash_functions(self):
+        """Create PostgreSQL-specific hash functions for Ibis expressions."""
+        import ibis
+
+        hash_functions = {}
+
+        @ibis.udf.scalar.builtin
+        def MD5(x: str) -> str:
+            """PostgreSQL MD5() function."""
+            ...
+
+        def md5_hash(col_expr):
+            return MD5(col_expr.cast(str))
+
+        hash_functions[HashAlgorithm.MD5] = md5_hash
+
+        @ibis.udf.scalar.builtin
+        def digest(value: str, algorithm: str) -> bytes:
+            """pgcrypto digest() function."""
+            ...
+
+        @ibis.udf.scalar.builtin
+        def encode(value: bytes, fmt: str) -> str:
+            """PostgreSQL encode() function."""
+            ...
+
+        @ibis.udf.scalar.builtin
+        def lower(value: str) -> str:
+            """PostgreSQL lower() function."""
+            ...
+
+        def sha256_hash(col_expr):
+            digest_expr = digest(col_expr.cast(str), ibis.literal("sha256"))
+            encoded = encode(digest_expr, ibis.literal("hex"))
+            return lower(encoded)
+
+        hash_functions[HashAlgorithm.SHA256] = sha256_hash
+
+        return hash_functions
+
+    @contextmanager
+    def open(self, mode: AccessMode = AccessMode.READ) -> Iterator[Self]:
+        """Open connection to PostgreSQL and perform capability checks.
+
+        Args:
+            mode: Access mode (READ or WRITE). Defaults to READ.
+
+        Yields:
+            Self: The store instance with connection open
+
+        Raises:
+            ImportError: If psycopg driver not installed.
+            Various database errors: If connection fails.
+        """
+        # Call parent context manager to establish connection
+        with super().open(mode):
+            try:
+                # Reset pgcrypto check for the new connection
+                self._pgcrypto_extension_checked = False
+                import sys
+
+                if not self._has_native_struct_support():
+                    self._struct_compat_mode = True
+                    message = (
+                        "!!! Metaxy WARNING: PostgreSQL backend lacks native STRUCT type support. "
+                        "Falling back to JSON serialization compatibility mode. !!!"
+                    )
+                    print(message, file=sys.stderr)
```

Comment:
What is that for? We already have the logger.warning below

Reply:
removed